.github/workflows/continuous_integration.yml

```diff
@@ -0,0 +1,170 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - dev
+      - rb/parallelize-ci-workflow
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build_and_publish_image:
+    name: Build and Publish Image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: pgc-images.sbgenomics.com/d3b-bixu/open-pedcan
+          tags: |
+            type=raw,value=analysisjob
+            # Only tag the image with latest if we're building on the default
+            # branch (e.g., dev).
+            type=raw,value=latest,enable={{is_default_branch}}
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          registry: pgc-images.sbgenomics.com
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+  run_analysis:
+    name: Run Analysis
+    needs: build_and_publish_image
```

Comment:
The `run_analysis` job depends on the `build_and_publish_image` job being successful.